addons/web_editor/controllers/main.py

```diff
@@ -223,11 +226,173 @@ def remove(self, ids, **kwargs):
+
+        return dict(
+            views = get_views and views or [],
+            less = get_less and less_datas or [],
+        )
+
+    ## The save_less route is in charge of saving a given modification of a LESS file.
+    ## @param url - the original url of the LESS file which has to be modified
+    ## @param bundle_xmlid - the xmlid of the bundle in which the LESS file addition can be found
+    ## @param content - the new content of the LESS file
+    @http.route("/web_editor/save_less", type="json", auth="user")
+    def save_less(self, url, bundle_xmlid, content):
+        IrAttachment = request.env["ir.attachment"]
+
+        custom_url = self._make_custom_less_file_url(url, bundle_xmlid)
+
+        # Check if the file to save had already been modified
+        custom_attachment = IrAttachment.search([("url", "=", custom_url)])
+        if custom_attachment:
+            # If it was already modified, simply override the corresponding attachment content
+            custom_attachment.write({"datas": content.encode("utf-8").encode("base64")})
+        else:
+            # If not, create a new attachment to copy the original LESS file content, with its modifications
+            IrAttachment.create(dict(
+                name = custom_url,
+                type = "binary",
+                mimetype = "text/less",
+                datas = content.encode("utf-8").encode("base64"),
+                datas_fname = url.split("/")[-1],
+                url = custom_url, # Having an attachment of "binary" type with an non empty "url" field
+                                  # is quite of an hack. This allows to fetch the "datas" field by adding
+                                  # a <link/> with the "url" content in the bundle template (see qweb)
+            ))
+
+            # Create a view to extend the template which adds the original file to link the new modified version instead
+            IrUiView = request.env["ir.ui.view"]
+            view_to_xpath = IrUiView.get_related_views(bundle_xmlid, bundles=True).filtered(lambda v: v.arch.find(url) >= 0)
+            IrUiView.create(dict(
+                name = custom_url,
+                mode = "extension",
+                inherit_id = view_to_xpath.id,
+                arch = """
+                    <data inherit_id="%(inherit_xml_id)s" name="%(name)s">
+                        <xpath expr="//link[@href='%(url_to_replace)s']" position="attributes">
+                            <attribute name="href">%(new_url)s</attribute>
+                        </xpath>
+                    </data>
+                """ % dict(
+                    inherit_xml_id = view_to_xpath.xml_id,
+                    name = custom_url,
+                    url_to_replace = url,
+                    new_url = custom_url,
+                )
+            ))
+
+            # TODO @chm, this works... sometimes
+            IrUiView.clear_caches()
+        IrAttachment.clear_caches()
```

Comment:
@Gorash Is that what you were thinking, it seems to work but not everytime... Could you take a look ?